### PR TITLE
Dashboard v2 current usage

### PR DESCRIPTION
**Pull request by meeh0w**
## Overview

* Fixes recent uploads URL (typo)
* Introduces `PlansProvider` -- this is because (at least) two of our components on the main screen need information about available subscription plans and the currently active plan for the user. Having it as a `Context` in the app allows us to minimize the number of requests we send.
* Adds a bit nicer loading indicators
* Fetches live data about the current usage

### CODE REVIEW

**Review by kwypchlo — APPROVED** [review]
looks good!

**Review by peterjan — CHANGES_REQUESTED** [review]

File: packages/dashboard-v2/src/components/CurrentUsage/CurrentUsage.js [hunk 1]
```diff
@@ -28,29 +50,43 @@ const size = (bytes) => {
...
+
+  if (loading) {
+    return <ContainerLoadingIndicator />;
+  }
 
-  const storageUsage = size(storage.used);
-  const storageLimit = size(storage.limit);
-  const filesUsedLabel = React.useMemo(() => ({ value: files.used, unit: "files" }), [files.used]);
+  if (error) {
+    return <ErrorMessage />;
```
[the remark this answers is not in the document]
Comment (meeh0w) [reply packages/dashboard-v2/src/components/CurrentUsage/CurrentUsage.js:71]: Nah, this component has a static error message. It's defined a few lines higher.

File: packages/dashboard-v2/src/components/CurrentUsage/CurrentUsage.js [hunk 2]
```diff
@@ -1,24 +1,46 @@
...
-  },
-});
+const useUsageData = () => {
+  const { user } = useUser();
+  const { activePlan, error } = useActivePlan(user);
+  const [loading, setLoading] = useState(true);
+  const { data: stats, error: statsError } = useSWR("user/stats");
+  const [usage, setUsage] = useState({});
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (meeh0w) [reply packages/dashboard-v2/src/components/CurrentUsage/CurrentUsage.js:20]: Good suggestion 👍

File: packages/dashboard-v2/src/components/LatestActivity/ActivityTable.js [hunk 3]
```diff
@@ -12,8 +13,8 @@ export default function ActivityTable({ type }) {
   if (!items.length) {
     return (
       <div className="flex w-full h-full justify-center items-center text-palette-400">
-        {/* TODO: proper loading indicator / error message */}
-        {!data && !error && <p>Loading...</p>}
+        {/* TODO: proper error message */}
+        {!data && !error && <ContainerLoadingIndicator />}
         {!data && error && <p>An error occurred while loading this data.</p>}
         {data && <p>No files found.</p>}
```
[the remark this answers is not in the document]
Comment (meeh0w) [reply packages/dashboard-v2/src/components/LatestActivity/ActivityTable.js]: Good spot! 74675f19

**Review by kwypchlo — APPROVED** [review]

File: packages/dashboard-v2/src/components/CurrentUsage/CurrentUsage.js [hunk 4]
```diff
@@ -1,24 +1,50 @@
...
+
+    if (hasData && !hasError) {
+      setUsage({
+        filesUsed: stats?.numUploads,
+        filesLimit: activePlan?.limits?.maxNumberUploads,
+        storageUsed: stats?.totalUploadsSize,
+        storageLimit: activePlan?.limits?.storageLimit,
+      });
+    }
+  }, [hasData, hasError, stats, activePlan]);
```
Comment (peterjan) [inline packages/dashboard-v2/src/components/CurrentUsage/CurrentUsage.js:37]: `nit` sorry for nitting on a change but just for my understand, `stats` and `activeplan` are redundant here right?

**Review by peterjan — APPROVED** [review]

File: packages/dashboard-v2/src/components/CurrentUsage/CurrentUsage.js (hunk 4 above)
Comment (meeh0w) [reply packages/dashboard-v2/src/components/CurrentUsage/CurrentUsage.js:37]: No :)

* `activePlan` contains information about the active subscription, including bandwidth and storage limits
* `stats` contain the current usage of those limits

**Pull request merged**